netplan/cli/utils.py

```diff
@@ -62,6 +65,40 @@ def systemctl_network_manager(action):  # pragma: nocover (covered in autopkgtes
     subprocess.check_call(['systemctl', action, '--no-block', service_name])
 
 
+def gather_replug_yaml(root_dir):
```

Comment:
@vorlonofportland  Could we have your take as architect on this particular schema change, and the PR in general?

Reply:
From a support point of view, I really want a kill switch. It will mean that I can ask users/customers who are experiencing issues to disable replug everywhere, and if that helps, to then narrow down a more specific rule. It means they have a work-around while we work on a fix.
From my point of view, trying to get a per-interface matching rule right - especially in the presence of renaming - is likely to require a few round-trips. If we don't have a workaround, it could well lead to lots of frustration. That's just where I'm coming from.

I'm happy for it to be something ominously named, like `disable_all_replug: yes_I_know_this_may_break_things`, if that helps.

Reply:
I don't think you understand my point; having:

network:
    ethernets:
        blah:
            replug: no

Would appropriately allow you to avoid replugging one interface -- there's no need to blanked disable everything.

Reply:
I'm still not clear on how that would work.

Say I have the following:
```
network:
  version: 2
  ethernets:
   ens8:
     match:
       name: ens8
     addresses: [10.10.10.2/24]
```

Now this will work without a replug if I don't rename ens8. But, if I do the following:
```
ubuntu@netplan:~$ sudo ip a del dev ens8 10.10.10.2/24
ubuntu@netplan:~$ sudo ip link set dev ens8 down
ubuntu@netplan:~$ sudo ip link set dev ens8 name renamedif0
```

The device now has a different name. Currently, it won't be matched based on the name in the first pass, so it will be replugged and then matched - per the following from `netplan --debug apply`:

```
DEBUG:replug renamedif0: unbinding virtio4 from /sys/bus/virtio/drivers/virtio_net
DEBUG:replug renamedif0: rebinding virtio4 to /sys/bus/virtio/drivers/virtio_net
```
(`ip -a` shows the match has worked - the interface has the IP address.)

So if I put `replug: no` in to the device stanza, it won't work: the match rule won't kick in until after the device has been replugged.

Reply:
That's expected behavior; the name of the interface doesn't match the match: clause. If you want renaming to work correctly, you need to positively, uniquely identify a device; the same applies to disabling replug.